node/bft/src/worker.rs

```diff
@@ -317,17 +324,26 @@ impl<N: Network> Worker<N> {
     ) -> Result<()> {
         // Construct the transmission.
         let transmission = Transmission::Solution(solution.clone());
+        // Compute the checksum.
+        let checksum = solution.to_checksum::<N>()?;
+        // Construct the transmission ID.
+        let transmission_id = TransmissionID::Solution(solution_id, checksum);
         // Remove the solution ID from the pending queue.
-        self.pending.remove(solution_id, Some(transmission.clone()));
+        self.pending.remove(transmission_id, Some(transmission.clone()));
         // Check if the solution exists.
-        if self.contains_transmission(solution_id) {
-            bail!("Solution '{}' already exists.", fmt_id(solution_id));
+        if self.contains_transmission(transmission_id) {
+            bail!("Solution '{}.{}' already exists.", fmt_id(solution_id), fmt_id(checksum));
```

Comment:
```suggestion
            bail!("Solution '{}.{}' already exists.", fmt_id(solution_id), fmt_id(checksum).dimmed());
```
nit (this is dimmed for `Transaction`)